### PR TITLE
Feature/messanging channel integration

### PR DESCRIPTION
**Pull request by johibkhan2**

### CODE REVIEW

**Review by admin-vizcentric — APPROVED** [review]

File: src/main/proto/Report.proto [hunk 1]
```diff
@@ -60,4 +65,24 @@ message Schedule {
 message ReportLog {
     string task_status = 1;
     string task_executed = 2;
+    string threshold_met = 3;
```
Comment (ir73) [inline src/main/proto/Report.proto:68]: I am not sure it's a good idea to have different coding style for messages. Some fields are camelCase, and some a re snake_case. I think we should use only one or another.

File: src/main/proto/Report.proto [hunk 2]
```diff
@@ -60,4 +65,24 @@ message Schedule {
...
+    string notification_sent = 4;
```
Comment (ir73) [inline src/main/proto/Report.proto:69]: could you please add an example what is a possible value for this? notification sent sounds like a boolean.

File: src/main/proto/Report.proto [hunk 3]
```diff
@@ -42,7 +42,12 @@ message AssignReport {
     string stride_API_Token = 4;
     string stride_cloud_id = 5;
     string stride_conversation_id = 6;
-    repeated Email email_list = 7;
+    CommunicationList communication_list = 7;
+}
+
+message CommunicationList{
+	repeated Email email = 1;
+	repeated int32 teams = 2;
```
Comment (ir73) [inline src/main/proto/Report.proto:50]: what is the possible value for this field?

File: src/main/proto/Report.proto [hunk 4]
```diff
@@ -60,4 +65,24 @@ message Schedule {
...
+    string threshold_met = 3;
+    string notification_sent = 4;
+    string channel = 5;
+}
+
+message ChannelConnection {
+    repeated Channelparameters ChannelProperties = 1;
+}
+
+message Channelparameters {
```
Comment (ir73) [inline src/main/proto/Report.proto:77]: should be camelCase for consistency `ChannelParameters`

File: src/main/proto/Report.proto [hunk 5]
```diff
@@ -60,4 +65,24 @@ message Schedule {
...
+  string id = 1;
+  repeated ConnectionProperties connectionProperties = 2;
+}
+
+message ConnectionProperties {
+    string displayName = 1;
```
Comment (ir73) [inline src/main/proto/Report.proto:83]: here we have camelCase field names for some reason.

File: src/main/proto/Report.proto (hunk 3 above)
Comment (khushbu-maurya) [reply src/main/proto/Report.proto:50]: > what is the possible value for this field?

for Email possible is user_email and user_name and for teams its integer array

File: src/main/proto/Report.proto (hunk 1 above)
Comment (khushbu-maurya) [reply src/main/proto/Report.proto:68]: Mostly we have snake_case in property name
use  camelCase when define custome message

File: src/main/proto/Report.proto (hunk 2 above)
Comment (khushbu-maurya) [reply src/main/proto/Report.proto:69]: yes its boolean I will correct it

File: src/main/proto/Report.proto (hunk 4 above)
Comment (khushbu-maurya) [reply src/main/proto/Report.proto:77]: okay

File: src/main/proto/Report.proto (hunk 5 above)
Comment (khushbu-maurya) [reply src/main/proto/Report.proto:83]: okay

File: src/main/proto/Report.proto (hunk 3 above)
Comment (ir73) [reply src/main/proto/Report.proto:50]: could you bring an example of a `teams` value? What do these integers represent?

File: src/main/proto/Report.proto (hunk 2 above)
Comment (ir73) [reply src/main/proto/Report.proto:69]: and then `threshold_met` as well should be bool?

**Pull request merged**